pkg/sdk/testint/basic_object_tracking_integration_test.go

```diff
@@ -0,0 +1,94 @@
+package testint
+
+import (
+	"context"
+	"fmt"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk"
+	"github.com/snowflakedb/gosnowflake"
+	"github.com/stretchr/testify/require"
+	"strings"
+	"testing"
+)
+
+// Research for basic object tracking done as part of SNOW-1737787
+
+// https://docs.snowflake.com/en/sql-reference/parameters#query-tag
+func TestInt_ContextQueryTags(t *testing.T) {
```

Comment:
what about the precedence test (i.e. set on ACC -> user -> session -> gosnowflake.WithQueryTag)? we are not going with this method now, so please leave a TODO with issue number at least

Reply:
A query tag is a regular parameter that goes under the regular hierarchy of precedence; the only nuance with this one is that we can set it on the query level, which was not documented. I tested it here by setting parameter on session level (the "lowest possible" one) to see if setting it on query level overrides it (it does). I'll describe it at the top, so it's clear.

Reply:
it would be nice to have at least one more level tested because then it's a clear answer when someone asks why not use query tags

Reply:
What about this one?

Reply:
added user level